src/components/templates/Jazzbar/components/JazzBarRoom/Room.tsx

```diff
@@ -160,29 +136,51 @@ const Room: React.FC<RoomProps> = ({
         }
         return prevParticipants.filter((p) => p !== participant);
       });
-    };
+    },
+    []
+  );
+
+  const connectToRoom = useCallback(() => {
+    if (!token) return;
 
-    Video.connect(token, {
+    return Video.connect(token, {
       name: roomName,
     })
       .then((room) => {
         setRoom(room);
-        localRoom = room;
         room.on("participantConnected", participantConnected);
         room.on("participantDisconnected", participantDisconnected);
         room.participants.forEach(participantConnected);
+
+        return room;
       })
       .catch((error) => setVideoError(error.message));
+  }, [roomName, token, participantConnected, participantDisconnected]);
 
-    return () => {
-      if (localRoom && localRoom.localParticipant.state === "connected") {
-        localRoom.localParticipant.tracks.forEach((trackPublication) => {
-          stopLocalTrack(trackPublication.track);
-        });
-        localRoom.disconnect();
-      }
-    };
-  }, [roomName, token, setParticipantCount]);
+  const reconnectToVideoRoom = useCallback(() => {
+    if (!token) return;
+
+    setVideoError("");
+
+    connectToRoom();
+  }, [connectToRoom, token]);
+
+  useEffect(() => {
+    const roomConnection = connectToRoom();
+
+    if (roomConnection) {
+      roomConnection.then((room) => {
+        return () => {
+          if (room && room.localParticipant.state === "connected") {
+            room.localParticipant.tracks.forEach((trackPublication) => {
+              stopLocalTrack(trackPublication.track);
+            });
```

Comment:
```suggestion
            room.localParticipant.tracks.forEach((trackPublication) => void stopLocalTrack(trackPublication.track) );
```

Might be simplified even more

```suggestion
            room.localParticipant.tracks.forEach( stopIt );
```

Reply:
Applied the first option in https://github.com/sparkletown/sparkle/pull/2647/commits/e96d5d0b1b518b09b0c335ad0710dc76ac006101